### ISSUE TITLE
Horizontal scrolling can occur in desktop

### ISSUE THREAD

**Issue by KernelDeimos** [HeyPuter/puter#710]
**Notes**
- **[good first issue]** - is tagged because this problem doesn't require much Puter-specific knowledge, but this issue in particular might be challenging to all but the CSS savant.

**Issue Description**
It is still possible to scroll horizontally on Puter when windows are dragged partially past the right side of the screen.

**Steps to Reproduce**
This is difficult to reproduce consistently but the following have worked *sometimes*:
- dragging an app window past the right-edge, which will then "bounce back", and then right-clicking the app icon in the titlebar.
- dragging an app window past the right-edge, then re-focusing it from the taskbar while another app is maximized

This was tested in Chromium

**Expected & Actual Behavior**
- body shouldn't scroll

**Addition Information or Screenshots (if applicable)**
![image](https://github.com/user-attachments/assets/e5117d60-0998-4990-bea6-24df2e6564a7)

**Deployment**
- [x] Production (puter.com)
- [ ] Development (`npm run start`)
- [ ] Docker (via `docker run`)
- [ ] Docker (via `docker-compose`)

**Puter version (if accessible)**
2.4.1

**Comment by Sarishc** [comment]
I'm interested  on working with this project

**Comment by jelveh** [comment]
awesome! assigned. Let me know if you need help 😇

**Comment by SaqAsh** [comment]
hey I am interested on working on this issue, I can see that this issue has not been worked on in a while

**Comment by jelveh** [comment]
Thank you! Go for it. Assigned.

**Comment by jelveh** [comment]
Let me know if you need any help

**Comment by SaqAsh** [comment]
Thanks! Probably gonna need it, first time here haha

**Comment by jelveh** [comment]
I'm here to help 🫡

**Comment by SaqAsh** [comment]
Hey! I was able to trigger this issue consistently.  All you need to do is have the screen be a size similar to one of a laptop, and follow one of the steps listed above. The difficulty is finding what css is causing this issue, currently looking at style.css in gui. Please let me know if you would be able to point me in a direction. That would be great! Thanks

**Comment by jelveh** [comment]
Ok, I'm a little confused now because I can't reproduce this anyway. Could you tell me how you managed to reproduce it consistently? If I can reprod I'll be able to help with the CSS

**Comment by SaqAsh** [comment]
To reproduce the issue, I followed these steps:

Moved the text editor to the right edge of the screen.
Maximized the Dev Center window to full screen.
Clicked the taskbar at the bottom to refocus on the text editor.
This reliably triggers the bug, but only when the window is resized to be smaller than the screen. I've attached a screenshot for reference.
Let me know if it works! 
![image](https://github.com/user-attachments/assets/069ce466-b5b9-4b60-b617-0157d2412a12)

**Comment by jelveh** [comment]
Yep, now I can reprod. Thank you for the explanation. I'm trying to figure out what causes it and will update you here when I find out. I suspect it has to do with how `focusWindow` works, but not sure yet.

**Comment by SaqAsh** [comment]
Yeah do let me know, that would be great!

**Comment by jelveh** [comment]
An eternity later, I think I know what the issue is now! It's the forced focus that is applied to the text input of notepad when the taskbar item is clicked. Now I need to figure out a fix.

**Comment by SaqAsh** [comment]
> An eternity later, I think I know what the issue is now! It's the forced focus that is applied to the text input of notepad when the taskbar item is clicked. Now I need to figure out a fix.

Haha let me know!

**Comment by prateekbisht23** [comment]
hey I am interested on working on this issue, can you assign this to me!!

**Comment by jelveh** [comment]
Awesome! assigned. Let us know if we could help with anything.

**Comment by prateekbisht23** [comment]
Is this the issue I'm looking at!?https://github.com/user-attachments/assets/73793a0a-8362-4430-b33c-43a353576fbd

**Comment by prateekbisht23** [comment]
the codebase is pretty big and its difficult to locate the part that causes the issue!!

**Comment by jelveh** [comment]
Yeah, this is a tricky one. I suspect multiple things are interacting with each other causing it. Where are you at right now in terms of clues? I should be able to help

**Comment by prateekbisht23** [comment]
From what I’ve seen so far, the desktop apps are created using a shared ID-based system rather than as standalone modules. This makes it tricky to isolate and work on the Notepad app specifically, since a lot of its behavior is coupled with the other apps!!
Also I’m also struggling to find where Notepad’s details (CSS, styling, and related logic) are defined, particularly the part that might be causing the problem. Any pointers on where I should be looking!?

**Comment by pranav200408** [comment]
Can I take this up?
